src/accelerate/accelerator.py

```diff
@@ -1809,6 +1814,25 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
                 model = torch.compile(model, **self.state.dynamo_plugin.to_kwargs())
         return model
 
+    def _build_device_mesh(self):
+        """Build and validate device mesh based on parallelism configuration."""
+        mesh_shape, mesh_dim_names = self.parallelism_config.get_mesh()
+    
+        if (existing_mesh := PartialState().device_mesh) is not None:
+            self.parallelism_config.validate_device_mesh(existing_mesh)
+            self.state.device_mesh = existing_mesh
+        else:
+                
+            device_mesh = torch.distributed.init_device_mesh(
+                self.device.type, 
+                mesh_shape=mesh_shape, 
+                mesh_dim_names=mesh_dim_names
+            )
+            if set(("dp_replicate", "dp_shard")).issubset(set(device_mesh.mesh_dim_names)):
+                device_mesh[("dp_replicate", "dp_shard")]._flatten("dp_fsdp")
```

Comment:
```suggestion
                device_mesh = device_mesh[("dp_replicate", "dp_shard")]._flatten("dp_fsdp")
```
I think you need to reassign since this doesn't update in place?